### PR TITLE
Adding Logging Support for Caught Exceptions

### PR DESCRIPTION
**Pull request by DhruvSondhi**
This PR aims to add logging support to caught exceptions.

**Description**

There are numerous exceptions that are caught but are not visible to the user. This PR aims to add support via logging for these exceptions such that it is visible when logging is turned on. Logs at `DEBUG` level 😄 

**Motivation and context**

Allows for better explanation for silent errors or exceptions that may happen. Logging them will benefit & make it more descriptive for the user to fix and see what is going on in the simulation. 

**How has this been tested?**
- [x] Testing pipeline.
- [x] Other.  Local Tests

**Type of change**

- [x] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [ ] My change requires a change to the documentation.
    - [ ] I have updated the documentation accordingly.
    - [ ] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: tardis/analysis.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
 
 import re
 import os
+import logging
```
Comment (andrewfullard) [inline tardis/analysis.py:5]: Why are there changes in this file? They don't do anything.

File: tardis/analysis/opacities.py [hunk 2]
```diff
@@ -329,7 +329,7 @@ def _calc_thomson_scattering_opacity(self):
         """Calculate the Thomson scattering opacity for each grid cell
 
         \[
-          \chi_{\mathrm{Thomson}} = n_{\mathrm{e}} \sigma_{\mathrm{T}}
+        \chi_{\mathrm{Thomson}} = n_{\mathrm{e}} \sigma_{\mathrm{T}}
```
Comment (andrewfullard) [inline tardis/analysis/opacities.py:332]: Why is this changed?

File: tardis/io/atom_data/base.py [hunk 3]
```diff
@@ -164,23 +164,33 @@ def from_hdf(cls, fname=None):
...
+                    logger.debug("\n\tDataframe does not contain NAME column")
                     nonavailable.append(name)
 
             atom_data = cls(**dataframes)
 
             try:
                 atom_data.uuid1 = store.root._v_attrs["uuid1"].decode("ascii")
             except KeyError:
+                logger.debug(
+                    "\n\tUUID not available for Atom Data\n\tSetting value to NONE"
```
Comment (andrewfullard) [inline tardis/io/atom_data/base.py:176]: I think NONE should be written as None to make it clear that it is the Python object None.

File: tardis/io/parsers/csvy.py [hunk 4]
```diff
@@ -37,6 +40,7 @@ def load_csvy(fname):
         try:
             data = pd.read_csv(fname, skiprows=yaml_end_ind + 1)
         except pd.errors.EmptyDataError as e:
+            logger.debug("\n\tCould not Read CSV.\n\tSetting Data to NONE")
```
Comment (andrewfullard) [inline tardis/io/parsers/csvy.py:43]: See previous comment. Perhaps specify which "data" this is?

File: tardis/montecarlo/base.py [hunk 5]
```diff
@@ -335,7 +335,7 @@ def virtual_packet_nu(self):
         try:
             return u.Quantity(self.virt_packet_nus, u.Hz)
         except AttributeError:
-            warnings.warn(
+            logger.debug(
```
Comment (andrewfullard) [inline tardis/montecarlo/base.py:338]: I think this should be a warning

File: tardis/montecarlo/base.py [hunk 6]
```diff
@@ -349,7 +349,7 @@ def virtual_packet_energy(self):
         try:
             return u.Quantity(self.virt_packet_energies, u.erg)
         except AttributeError:
-            warnings.warn(
+            logger.debug(
```
Comment (andrewfullard) [inline tardis/montecarlo/base.py:352]: Same here

File: tardis/montecarlo/base.py [hunk 7]
```diff
@@ -363,7 +363,7 @@ def virtual_packet_luminosity(self):
         try:
             return self.virtual_packet_energy / self.time_of_simulation
         except TypeError:
-            warnings.warn(
+            logger.debug(
```
Comment (andrewfullard) [inline tardis/montecarlo/base.py:366]: Same here

File: tardis/analysis/opacities.py (hunk 2 above)
Comment (DhruvSondhi) [reply tardis/analysis/opacities.py:332]: Just for formatting consistency with Docstrings 😄

File: tardis/analysis.py (hunk 1 above)
Comment (DhruvSondhi) [reply tardis/analysis.py:5]: I wanted to ask about this. Do we need to log exceptions for this part of the source code 🤔

**Comment by wkerzendorf** [comment]
I think that is a nice start to things. First of all - in general this PR seems to contain a variety of different fixes and a lot of files. In general, fewer PRs targeted to specific issues is better. 
My main other question is there are `\n\t` in front of essentially all debug statements. why?

**Comment by DhruvSondhi** [comment]
> My main other question is there are `\n\t` in front of essentially all debug statements. why?

Hello @wkerzendorf, This is actually due to how the formatting for the logger has been done. Continuing from #1632, I have added a new line & some tab spaces between the messages such that it is easier to read. If we remove this (`\n\t`) then the message is on the same line, without any space, making it difficult to read (what is the problem or where is the problem because everything is squashed in a single line with line wrapping, etc problems). In the end, it is just for formatting purposes. 
If you think that the formatting is unnecessary, I can revert them as they used to be (formatting wise) :smile:

**Comment by andrewfullard** [comment]
Now that you have a fix for the `/n/t`, you can implement it here

**Comment by DhruvSondhi** [comment]
I have implemented it in PR #1704. I think we can merge this as it is here & then I can resolve this in that PR 😄

**Review by wkerzendorf — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]
